source/style-guide/style/version.txt

```diff
@@ -0,0 +1,140 @@
+
+* Features changed in an EOL release represent preexisting behavior in all 
+  supported releases.
+
+
+Use the following guidelines to reference product versions so that
+you can more easily find these references when the release reaches EOL:
+
+* Use version directives whenever possible.  The version directives make it
+  easier to locate on the page and in the {+rst+}:
+
+  .. code-block:: rst
+
+     .. versionadded:: 6.0
+
+  If a version reference needs more explanation, add the content 
+  to the directive block:
+
+  .. code-block:: rst
+
+     .. versionchanged:: 4.4
+
+        This command returns the following output:
+
+  If you need to call out multiple versions, set the most recent version
+  first, then reference the others in a phrase afterwards.
+
+  .. code-block:: rst
+
+     .. versionadded:: 6.0 *(Also available in MongoDB 5.0.)*
+
+  The italics marker matches the directive output.
+
+  Snooty supports the following version directives: 
+
+  .. hlist::
+     :columns: 3
+
+     * |add|
+     * |change|
+     * |deprecated|
+
+* To refer to a version inline, include the product
+  name before the version number, (for example, ``MongoDB 5.0``).  This ensures 
+  that a search for version numbers correctly matches product and release.
+
+* Do **not** reference versions freeform or with the phrase ``version X.X``.
+  These references are much harder to identify programmatically as references 
+  to a specific product version.
+
+Identify EOL Releases
+---------------------
+
+You can find references to EOL releases of MonogDB using grep:
+
+.. code-block:: shell
+
+   $ grep -RI \
+         -e ".. versionadded:: [0-9]" \
+         -e ".. versionchanged:: [0-9]" \
+         -e ".. deprecated:: [0-9]" \
+         -e "MongoDB [0-9]" \
+         source | \
+         grep -v "6\.2" | \
+         grep -v "6\.1" | \
+         grep -v "6\.0" | \
+         grep -v "5\.3" | \
+         grep -v "5\.1" | \
+         grep -v "5\.0" | \
+         grep -v "4\.4" | \
+         grep -v "4\.2" | \
+         grep -v "source/release-notes"
+   source/sharding.txt:128:- In MongoDB 4.0 and earlier, a document's shard key field value is
+   source/sharding.txt:351:   Starting in MongoDB 4.0.3, setting up zones and zone ranges *before*
+   source/sharding.txt:379:Starting in MongoDB 3.6, :doc:`change streams </changeStreams>` are
+   ...
+
+
+This command could be added to a script and adjusted to work from an
+arbitrary directory:
+
+.. code-block:: bash
+
+   grep -RIne ".. versionadded:: [0-9]" -e ".. versionchanged:: [0-9]" \
+         -e ".. deprecated:: [0-9]" -e "MongoDB [0-9]" $1 | \
+         grep -v "6\.2" | \
+         grep -v "6\.1" | \
+         grep -v "6\.0" | \
+         grep -v "5\.3" | \
+         grep -v "5\.1" | \
+         grep -v "5\.0" | \
+         grep -v "4\.4" | \
+         grep -v "4\.2" | \
+         grep -v "source/release-notes"
+
+To execute the script, call it with a path argument:
```

Comment:
Suggestion:

I would omit this subsection on converting it to a script since it's mostly duplicate, and converting a shell command to a script can be learned via google.

There also seems to be some minor details missing on preparing the script. For example, I think you'll need to set execution permissions and to use a shebang interpreter directive to run the command as listed. It's also not apparent the script should be called "version.sh" nor that the "source" argument is the name of a relative directory.

Reply:
Fixed